### ISSUE TITLE
CyclicalReference error when $VAR is undefined

### ISSUE THREAD

**Issue by rsgalloway** [rsgalloway/envstack#18]
when $VAR is undefined in the environment but is referenced in the env stack file, envstack throws a CyclicalReference error in 0.7.0:

```bash
$ cd env
$ envstack -r
...
envstack.exceptions.CyclicalReference: Cyclical reference detected in ${ROOT}/${ENV}
```

this can be resolved by giving $VAR a value in the environment, or by giving it a default null value when expanding.

**Comment by rsgalloway** [comment]
this is easy to reproduce. if we define an VAR like this:

```yaml
FOO: bar:${FOO}
```

then we can recreate the error:

```bash
$ envstack -r
...
envstack.exceptions.CyclicalReference: Cyclical reference detected in bar:${FOO}
```

**Comment by rsgalloway** [comment]
addressed in version 0.7.1
